src/transports/ipc.js

```diff
@@ -17,6 +19,13 @@ function getIPCPath(id) {
   if (process.platform === 'win32') {
     return `\\\\?\\pipe\\discord-ipc-${id}`;
   }
+  if (process.platform === 'darwin') {
+    let tmpdir = os.tmpdir();
```

Comment:
I think you can use [**child_process**](https://nodejs.org/api/child_process.html) to find out where the path to **discord-ipc** is. I don't know if it's the best solution, but it's worth a try. 

```js
const { exec } = require('child_process');

 if (process.platform === 'darwin') {
        exec(`find /private/var/folders -name "*discord-ipc*" 2> /dev/null`, (error, stdout, stderr) => {
            if (error || stderr) {
                return console.log(`error: ${error.message}`);
            }

            if (stderr) {
                return console.log(`stderr: ${stderr}`);
            }

            if(stdout === "") {
                return console.log(`fail.`);
            }

            // should return "/private/var/folders/g3/znz8yyl55ls27thh9l56x6rr0000gn/T/discord-ipc-0" 
            // or something like that.
            return stdout;
        });
    }
```

Reply:
Or I think it's the same think you did above. 😂

Reply:
So I get the temp directory, which ordinarily is enough however sometimes we're in the directory for an app (e.g. if we're running under a subprocess) hence I additionally make sure that it's in the parent `/T` if it isn't.

I recommend your method as a more complete version if mine doesn't work for all cases as I am relying on the assumption that a) the root is always in `/T` and b) we're never more than one directory down, however from what I understand those assumptions are always true, in which case, mine is a little more efficient as it doesn't need to search all the directories.